tests/dualstack_test.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2024 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Final, List
+
+from absl import flags
+from absl.testing import absltest
+
+from framework import xds_k8s_flags
+from framework import xds_k8s_testcase
+from framework.helpers import skips
+from framework.test_app.runners.k8s import k8s_xds_server_runner
+
+logger = logging.getLogger(__name__)
+flags.adopt_module_key_flags(xds_k8s_testcase)
+
+# Type aliases
+_XdsTestServer = xds_k8s_testcase.XdsTestServer
+_XdsTestClient = xds_k8s_testcase.XdsTestClient
+_KubernetesServerRunner = k8s_xds_server_runner.KubernetesServerRunner
+_Lang = skips.Lang
+
+_SERVERS_APP_LABEL: Final[str] = "psm-interop-dualstack"
+
+
+class DualStackTest(xds_k8s_testcase.RegularXdsKubernetesTestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Force the canonical test server for all languages."""
+        super().setUpClass()
+        if cls.lang_spec.client_lang is not _Lang.JAVA:
+            cls.server_image = xds_k8s_flags.SERVER_IMAGE_CANONICAL.value
+
+    def setUp(self):
+        self.enable_dualstack = True
```

Comment:
instead, let's assert it at the beginning of the test. This may prevent us from accidentally running it without `common-dualstack.cfg`.

Reply:
Done